tests/integration/queue-listeners-test.js

```diff
@@ -0,0 +1,89 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render, settled } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+import { selectFiles } from 'ember-file-upload/test-support';
+import { tracked } from '@glimmer/tracking';
+
+module('Integration | queue listeners', function (hooks) {
+  setupRenderingTest(hooks);
+
+  const setupState = function () {
+    const fileQueueService = this.owner.lookup('service:file-queue');
+
+    class State {
+      @tracked firstComponent = true;
+    }
+
+    this.state = new State();
+    this.queue = fileQueueService.create('listener-test');
+  };
+
+  const assertOnlyRenderedComponentCallbacks = async function (assert) {
+    await selectFiles('input[type=file', new File([], 'first.txt'));
+    this.state.firstComponent = false;
+    await settled();
+    await selectFiles('input[type=file', new File([], 'second.txt'));
```

Comment:
```suggestion
    await selectFiles('input[type=file]', new File([], 'second.txt'));
```

my inner monk wants the square bracket to be closed 😂

Reply:
Good catch 🙇🏻